setup

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# -e (-o errexit): exit immediately on error
+# -u (-o nounset): treat unset variables as erros
+# -o pipefail: propagate intermediate errors in pipelines
+set -euo pipefail
+
+main() {
```

Comment:
Bash requires functions to be defined before use, so when I write scripts with functions I put the most important part into a `main` function at the top and declare helpers below.